### PR TITLE
Add nft owner check + tx error log

### PR DESCRIPTION
**Pull request by mikiquantum**
<This is intended as a guide when describing-reviewing pull requests for go-centrifuge>
Issue Link: <Github link>

### CODE REVIEW

File: nft/ethereum_payment_obligation.go [hunk 1]
```diff
@@ -204,6 +204,17 @@ func (s *ethereumPaymentObligation) minter(ctx context.Context, tokenID TokenID,
 			return
 		}
 
+		// Check if tokenID exists in registry and owner is deposit address
+		owner, err := s.OwnerOf(req.RegistryAddress, tokenID[:])
```
Comment (manuelpolzhofer) [inline nft/ethereum_payment_obligation.go]: Just out of curiosity.
What happens if the user passes an invalid address as depositAddress?
Would it fail?

File: transactions/txv1/manager.go [hunk 2]
```diff
@@ -92,6 +92,7 @@ func (s *manager) ExecuteWithinTX(ctx context.Context, accountID identity.DID, e
 			if e == nil && transactions.TxIDEqual(existingTxID, transactions.NilTxID()) {
 				tempTx.Status = transactions.Success
 			} else if e != nil {
+				tempTx.Logs = append(tempTx.Logs, transactions.NewLog(desc, e.Error()))
```
Comment (vimukthi-git) [inline transactions/txv1/manager.go]: could you add something here that says log was added by `manager`?

File: nft/ethereum_payment_obligation.go (hunk 1 above)
Comment (mikiquantum) [reply nft/ethereum_payment_obligation.go]: good question, I think there must be a test case in truffle project that should verify that.

Comment (rdinicut) [reply nft/ethereum_payment_obligation.go]: What does invalid mean? There is a check for a valid ETH address.

Comment (rdinicut) [reply nft/ethereum_payment_obligation.go]: Valid means it passes a checksum test

Comment (lucasvo) [reply nft/ethereum_payment_obligation.go]: There are no checksum tests for addresses, the checksum is only in capitalization which is not really enforced outside of user input. Theoretically any 20 byte value is correct.

**Comment by lucasvo** [comment]
Is it possible to add unit tests here for the new code?

File: nft/ethereum_payment_obligation.go (hunk 1 above)
Comment (rdinicut) [reply nft/ethereum_payment_obligation.go]: Yes, but you can also enforce the checksum to have some extra validation. Web3 does that for example

**Pull request merged**